File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/DecimalToOctal/DecimalToOctal_test.js
import assert from 'assert'
import solution from './DecimalToOctal.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 299

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('453', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('1', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 2924

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('5554', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 94829

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('271155', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 342

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('526', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 2357182

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('10773676', result)
    done();
  });
})

File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/DecimalToOctal/DecimalToOctal_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.DecimalToOctal.DecimalToOctal import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 299

        # when
        result = solution(n)

        # then
        self.assertEqual('453', result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual('1', result)

    def test_3(self):
        # given
        n = 2924

        # when
        result = solution(n)

        # then
        self.assertEqual('5554', result)

    def test_4(self):
        # given
        n = 94829

        # when
        result = solution(n)

        # then
        self.assertEqual('271155', result)

    def test_5(self):
        # given
        n = 342

        # when
        result = solution(n)

        # then
        self.assertEqual('526', result)

    def test_6(self):
        # given
        n = 2357182

        # when
        result = solution(n)

        # then
        self.assertEqual('10773676', result)


if __name__ == '__main__':
    unittest.main()
